coderd/healthcheck/healthcheck.go

```diff
@@ -0,0 +1,37 @@
+package healthcheck
+
+import (
+	"context"
+	"time"
+
+	"golang.org/x/xerrors"
+	"tailscale.com/tailcfg"
+)
+
+type Report struct {
```

Comment:
Ideally there's an overall "Pass" or "Fail" grade, as well as a "Pass" or "Fail" for each of the sub-reports.

Reply:
I thought about that but at least for DERP pass/fail felt more like a spectrum. You could have a derp server that could exchange messages but was listed as STUN but STUN wasn't working, etc

Reply:
Eh maybe "Fail" could just mean anything failed or is degraded.